app/assets/javascripts/discourse/tests/unit/lib/autocomplete-test.js

```diff
@@ -1,154 +1,85 @@
+import { settled } from "@ember/test-helpers";
 import { setupTest } from "ember-qunit";
 import { compile } from "handlebars";
 import $ from "jquery";
 import { module, test } from "qunit";
-import autocomplete from "discourse/lib/autocomplete";
+import { simulateKey } from "discourse/tests/helpers/qunit-helpers";
 
 module("Unit | Utility | autocomplete", function (hooks) {
   setupTest(hooks);
   let elements = [];
 
-  function textArea(value) {
-    let element = document.createElement("TEXTAREA");
+  const template = compile(
```

Comment:
We're always using the same `template` so I extracted it so we only compile it once.